R/tree_accessors.R

```diff
@@ -2307,6 +2364,305 @@ setMethod(
+  ## we don't have intermediary structures in the coltree model
+  ## **after the first one** Yes this is bad and wrong but
+
+  ## its how it is.
+  ## ie it goes straight from A: Drug X -> F
+  ## in the split_cols_by("ARM") %>% split_cols_by("SEX") case
+  ## but we want to use the do_recursive_replace machinery
+  ## that is already well tested
+  trimmed_path <- path[c(1, seq(2, length(path), by = 2))]
+
+  do_recursive_replace(obj, trimmed_path, value = value)
+}
+
+
+#' Set visibility of column counts for a group of sibling facets
+#'
+#' @inheritParams gen_args
+#' @param path character. The path *to the parent of the
+#' desired siblings*. The last element in the path should
+#' be a split name.
+#' @return obj, modified with the desired columncount
+#' display behavior
+`facet_colcounts_visible<-` <- function(obj, path, value) {
+  coldf <- make_col_df(obj, visible_only = FALSE)
+  allpaths <- coldf$path
+  lenpath <- length(path)
+  match_paths <- vapply(allpaths, function(path_i) {
+    (length(path_i) == lenpath + 1) &&
+      (all(head(path_i, -1) == path))
+  }, TRUE)
+  for (curpath in allpaths[match_paths]) {
+    colcount_visible(obj, curpath) <- value
+  }
+  obj
+}
+
+#' Get or set column count for a facet in column space
+#'
+#' @inheritParams gen_args
+#' @param path character. This path must end on a
+#' split value, e.g., the level of a categorical variable
+#' that was split on in column space, but it need not
+#' be the path to an individual column.
+#'
+#' @return for `facet_colcount` the current count associated
+#' with that facet in column space, for `facet_colcount<-`,
+#' `obj` modified with the new column count for the specified
+#' facet.
+#'
+#' @note Updating a lower-level (more specific)
+#' column count manually **will not** update the
+#' counts for its parent facets. This cannot be made
+#' automatic because the rtables framework does not
+#' require sibling facets to be mutually exclusive
+#' (e.g., total "arm", faceting into cumulative
+#' quantiles, etc) and thus the count of a parent facet
+#' will not always be simply the sum of the counts for
+#' all of its children.
+#'
+#' @export
+#' @examples
+#' lyt <- basic_table() %>%
+#'   split_cols_by("ARM", show_colcounts = TRUE) %>%
+#'   split_cols_by("SEX", split_fun = keep_split_levels(c("F", "M")),
+#'     show_colcounts = TRUE) %>%
+#'   split_cols_by("STRATA1", show_colcounts = TRUE) %>%
+#'   analyze("AGE")
+#'
+#' tbl <- build_table(lyt, ex_adsl)
+#'
+#' facet_colcount(tbl, c("ARM", "A: Drug X"))
+#' facet_colcount(tbl, c("ARM", "A: Drug X", "SEX", "F"))
+#' facet_colcount(tbl, c("ARM", "A: Drug X", "SEX", "F", "STRATA1", "A"))
+#'
+#' ## modify specific count after table creation
+#' facet_colcount(tbl, c("ARM", "A: Drug X", "SEX", "F", "STRATA1", "A")) <- 25
+#'
+#' ## show black space for certain counts by assign NA
+#'
+#' facet_colcount(tbl, c("ARM", "A: Drug X", "SEX", "F", "STRATA1", "C")) <- NA
+setGeneric("facet_colcount",
```

Comment:
facet makes sense but it could be difficult to search for. Maybe `split_colcount` is more intuitive for users that are already using {rtables}

Reply:
splits don't have column counts, because splits are pre-data. once it is instantiated it is a facet. If users don't understand what facet means that is an education failure on our part but split_colcount isn't correct in the technical sense.

Reply:
Also I'd think people would be searching for "colcount" to try and figure out how to do this, in which case this would come up